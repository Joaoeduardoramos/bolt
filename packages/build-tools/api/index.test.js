// const { twigRenderer } = require('./twig-renderer');
const buttonTemplate = '@bolt-components-button/button.twig';


describe.skip('Test the Bolt TwigRender API', () => {
  test('renders a basic button', async () => {
    const buttonData = {
      'text': 'Hello world!',
    };
    const result = await twigRenderer(buttonTemplate, buttonData);

    expect(result.ok).toEqual(true);
  });


  test('handles missing data', async () => {
    const result = await twigRenderer(buttonTemplate);

    expect(result.ok).toEqual(true);
  });


  test('handles non-existent template path', async () => {
    const result = await twigRenderer(
      '@bolt-components-button/button2.twig',
      {
        'text': 'Hello world 2!',
      },
    );

    expect(result.ok).toEqual(false);
  });


  test('renders the button component correctly', async () => {
    const result = await twigRenderer(
      buttonTemplate,
      {
        'text': 'Hello world!2',
      },
    );

    expect(result.html).toMatchSnapshot();
  });


  test('renders attributes on the button component correctly', async () => {
    const result = await twigRenderer(
      buttonTemplate, {
        'text': 'Hello world!',
        'attributes': {
          'class': [
            'u-bolt-margin-bottom-large',
          ],
        },
      },
    );

    expect(result.html).toMatchSnapshot();
  });


  test('renders the secondary button correctly', async () => {
    const result = await twigRenderer(
      buttonTemplate,
      {
        'text': 'Secondary Button!',
        'style': 'secondary',
      },
    );

    expect(result.html).toMatchSnapshot();
  });
});