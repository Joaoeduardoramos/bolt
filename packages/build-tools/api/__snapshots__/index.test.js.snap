// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Bolt TwigRender API renders attributes on the button component correctly 1`] = `
"<bolt-button
     align=\\"center\\"      color=\\"primary\\"      size=\\"medium\\"                      transform=\\"none\\"     
        
    

            

                  >


    <button  class=\\"u-bolt-margin-bottom-large c-bolt-button__root\\">

        <replace-with-children class=\\"c-bolt-button  c-bolt-button--medium   c-bolt-button--primary c-bolt-button--none c-bolt-button--center\\">
      
      <span class=\\"c-bolt-button__item \\">Hello world!</span>

          </replace-with-children>

  </button>
</bolt-button>"
`;

exports[`Test the Bolt TwigRender API renders the button component correctly 1`] = `
"<bolt-button
     align=\\"center\\"      color=\\"primary\\"      size=\\"medium\\"                      transform=\\"none\\"     
        
    

            

                  >


    <button  class=\\"c-bolt-button__root\\">

        <replace-with-children class=\\"c-bolt-button  c-bolt-button--medium   c-bolt-button--primary c-bolt-button--none c-bolt-button--center\\">
      
      <span class=\\"c-bolt-button__item \\">Hello world!</span>

          </replace-with-children>

  </button>
</bolt-button>"
`;

exports[`Test the Bolt TwigRender API renders the secondary button correctly 1`] = `
"<bolt-button
     align=\\"center\\"      color=\\"secondary\\"      size=\\"medium\\"                      transform=\\"none\\"     
        
    

            

                  >


    <button  class=\\"c-bolt-button__root\\">

        <replace-with-children class=\\"c-bolt-button  c-bolt-button--medium   c-bolt-button--secondary c-bolt-button--none c-bolt-button--center\\">
      
      <span class=\\"c-bolt-button__item \\">Secondary Button!</span>

          </replace-with-children>

  </button>
</bolt-button>"
`;
